tests/flow/test_optimizations_plan.py

```diff
@@ -426,3 +426,31 @@ def test27_optimize_cartesian_product_complex_filter_trees(self):
                     [0, 3],
                     [0, 3]]
         self.env.assertEqual(resultset, expected)
+
+    # Label_id of a label in a cached execution plan should be updated properly.
+    # The label with less nodes should be traversed first.
+    def test28_optimize_label_scan_cached_label_id(self):
```

Comment:
Test intro comment:
`in cases where a referred label doesn't exists, the UNKNOW_LABEL_ID is being cached
once the label is created we want to make sure that the UNKNOW_LABEL_ID is replaced with the actual label ID
this test illustrates this scenario by traversing from a non-existing label (populating our execution-plan cache) which afterwards is being created, once created we want to make sure the correct label ID is used.`